docs/intro/fastapi_gelai_searchbot.rst

```diff
@@ -0,0 +1,1556 @@
+.. _ref_guide_fastapi_gelai_searchbot:
+
+===================
+FastAPI (Searchbot)
+===================
+
+:edb-alt-title: Building a searchbot with memory using FastAPI and Gel AI
+
+In this tutorial we're going to walk you through building a chat bot with search
+capabilities using Gel and `FastAPI <https://fastapi.tiangolo.com/>`_.
+
+FastAPI is a framework designed to help you build a web app, well, fast. And Gel
+is a data layer designed to help you figure out storage in your application, and
+also do it fast. By the end of this tutorial you will have tried out different
+aspects of using those two together.
+
+We're going to start by creating an app with FastAPI, then add web search
+capabilities to it, and then put search results through a language model to get
+a human-friendly answer. After that we'll use Gel to implement chat history, and
+finish it off with semantic search, so that the bot can remember previous
+interactions with the user.
+
+The end result is going to look something like this:
+
+.. image::
+    /docs/tutorials/placeholder.png
+    :alt: Placeholder
+    :width: 100%
+
+1. Initialize the project
+=========================
+
+We're going to start by installing `uv <https://docs.astral.sh/uv/>`_ - a Python
+package manager that's going to simplify environment management for us. You can
+follow their `installation instructions
+<https://docs.astral.sh/uv/getting-started/installation/>`_ or simply run:
+
+.. code-block:: bash
+
+    $ curl -LsSf https://astral.sh/uv/install.sh | sh
+
+Once that is done, we can use uv to create scaffolding for our project following
+the `documentation <https://docs.astral.sh/uv/guides/projects/>`_:
+
+.. code-block:: bash
+
+    $ uv init searchbot \
+      && cd searchbot
+
+For now, we know we're going to need Gel and FastAPI, so let's add those
+following uv's instructions on `managing dependencies
+<https://docs.astral.sh/uv/concepts/projects/dependencies/#optional-dependencies>`_,
+as well as FastAPI's `installation docs
+<https://fastapi.tiangolo.com/#installation>`_. Running ``uv sync`` after that
+will create our virtual environment in a ``.venv`` directory and ensure it's
+ready. As a last step, we'll activate the environment and get started.
+
+.. code-block:: bash
+
+    $ uv add "fastapi[standard]" \
+      && uv add gel \
+      && uv sync \
+      && source .venv/bin/activate
+
+.. note::
+
+    Make sure to source the environment every time you open a new terminal
+    session before running ``python``, ``gel`` or ``fastapi``-related commands.
+
+
+2. Get started with FastAPI
+===========================
+
+At this stage we need to follow FastAPI's `tutorial
+<https://fastapi.tiangolo.com/tutorial/>`_ to create the foundation of our app.
+
+We're going to make a super simple web API with one endpoint that takes in a
```

Comment:
In general, I'd avoid "simple", "easy", etc, as it can be discouraging for beginners (easy for us != easy for other people). 

```suggestion
We're going to make a minimal web API with one endpoint that takes in a
```